graphistry/layout/poset.py

```diff
@@ -0,0 +1,151 @@
+from collections import OrderedDict
+
+
+class Poset(object):
```

Comment:
A little surprising to see here -- sounds like it's primarily here for better speed on just __cmp__ / __eq__ ?

Reply:
A poset, seen as so-called Hasse diagram, is a layered layout without the multi-layer edges. In partially order language, if you remove the edges representing transitive closures you end up with a poset representing the layout you are after. Said differently, the Sugiyama layout is an algorithm which tries to find a partial order fitting a given pre-order. So, the poset structure embodies the layout in an algebraic thing. The cmp and all that is the partial order. You can implement the layout without a poset but the  operations (e.g. comparing where a vertex sits in the stack of layers wrt to another vertex) become less elegant. Layout algorithms are in general quite abstract and it helps to define things like this to support the conceptual flow.

Reply:
Yes, so I was wondering if we can reuse someone else's implementation, or even need it. It is a generic class and adds maintenance burden - a big file in this commit

Reply:
The Poset class is ~140 lines of code. You would like to use an external poset package to replace this file?

Reply:
If there is something reasonable; custom collections is a bad idea if can be avoided